flow-typed/npm/@react-native-clipboard/clipboard_vx.x.x.js

```diff
@@ -0,0 +1,63 @@
+// flow-typed signature: 12bef3bad479f254c9e8da5a51023c58
```

Comment:
From the commit message:

```
Also,this library has built-in Flow support, so we don't need to make our
own libdef.
```

Is this true? I see that you've added a stub libdef with no types in it. If you delete that, do we get good type-checking (i.e., Flow correctly complains when you try to do something wrong)?

Reply:
@chrisbobbe Thanks for review!

Hmm... I remember now , the problem occured was that this error `Importing from an untyped module makes it any and is not safe! Did you mean to add // @flow to the top of @react-native-clipboard/clipboard?Flow(untyped-import)`  always occured 
And after some research I found I have to add a stub libdef for this(which Ithink it should be optional here :thinking:)

Reply:
OK; that sounds like a sign that "Also,this library has built-in Flow support" is wrong. 🙂 Is there still something that leads you to believe that the library has built-in Flow support?

As mentioned above, if it turns out that it doesn't come with types, we'll have to write our own with a meaningful libdef.

Reply:
OK , now I get it...seems earlier I misunderstood the built-in Flow support thing pretty much :( .
> As mentioned above, if it turns out that it doesn't come with types, we'll have to write our own with a meaningful libdef.

I'm a little confused about this,
AFAIK `@react-native-clipboard/clipboard` has published libdef available(so we can get autogenerated libdef stub for it) , 
Or do we have to write separate libdef mentioning proper types , for e.g, we have for `react-native-webview`

Reply:
Ah, OK, so the thing we usually do to see if there's a libdef published by `flow-typed` is this:

```
flow-typed install @react-native-clipboard/clipboard
```

I've tried this command just now, and it gave this output:

```
[@octokit/rest] `const Octokit = require("@octokit/rest")` is deprecated. Use `const { Octokit } = require("@octokit/rest")` instead
• Searching for 1 libdefs...
• rebasing flow-typed cache...
!! No flow@v0.128.0-compatible libdefs found in flow-typed for the explicitly requested libdefs. !!

Consider using `flow-typed create-stub @react-native-clipboard/clipboard` to generate an empty libdef that you can fill in.
```

(I have no idea what the `@octokit/rest` thing is about; ignoring that.)

Notice the "No […] libdefs found in flow-typed […]."

The file you made looks like it was created by running the recommended command, `flow-typed create-stub @react-native-clipboard/clipboard`. But, as the rest of that sentence says (and as the generated file says), the file you get from that is basically empty; there's some boilerplate but nothing substantive. It won't have useful types until we write some. 🙂

The libdef for `react-native-webview` is an example where we wrote one, but I expect this one would be much easier than that—we use just one function it provides, `Clipboard.setString`, and that function's type seems pretty simple. Its parameters are described [here](https://github.com/react-native-clipboard/clipboard#setstring), and I think it probably returns `undefined`.